src/org/opensolaris/opengrok/history/GitRepository.java

```diff
@@ -252,11 +246,154 @@ public InputStream getHistoryGet(String parent, String basename, String rev) {
                 }
             }
         }
+        return ret;
+    }
+
+    @Override
+    public InputStream getHistoryGet(String parent, String basename, String rev) {
+        String fullpath;
+        try {
+            fullpath = new File(parent, basename).getCanonicalPath();
+        } catch (IOException exp) {
+            LOGGER.log(Level.SEVERE,
+                    "Failed to get canonical path: {0}", exp.getClass().toString());
+            return null;
+        }
+
+        InputStream ret = getHistoryRev(fullpath, rev);
+        if (ret == null) {
+            /*
+             * If we failed to get the contents it might be that the file was
+             * renamed so we need to find its original name in that revision
+             * and retry with the original name.
+             */
+            String origpath;
+            try {
+                origpath = findOriginalName(fullpath, rev);
+            } catch (IOException exp) {
+                LOGGER.log(Level.SEVERE,
+                        "Failed to get original revision: {0}",
+                        exp.getClass().toString());
+                return null;
+            }
+            if (origpath != null) {
+                ret = getHistoryRev(origpath, rev);
+            }
+        }
 
         return ret;
     }
 
     /**
+     * Expand format from the git output.
+     * <br>
+     * from:<br>
+     * moved/{renamed.c => renamed2.c} (100%)<br>
+     * to:<br>
+     * moved/renamed.c => moved/renamed2.c (100%)<br>
+     *
+     * @param renamedFile the string to expand
+     * @return normalized string
+     */
+    protected String expandGitRenamedOutput(String renamedFile) {
+        // the pattern should cover format like this:
+        // moved/{renamed.c => renamed2.c} (100%)
+        Pattern pattern = Pattern.compile("\\s*([^\\{]+)\\s*\\{\\s*(.+?)\\s+=>\\s+([^\\}]+)\\s*\\}(.*)");
```

Comment:
I forgot to mention that I changed the command to obtain more predictable output format (without these `{, }` constructs. So this method is no longer available :-)